### ISSUE TITLE
__builtins__ and __loader__ are not recognized as builtins

### ISSUE THREAD

**Issue by tweakimp** [PyCQA/pylint#3441]
### Steps to reproduce
```python
print(__builtins__)
print(__loader__)
```
Save and run. It works. `__builtins__` and `__loader__` are defined.

### Current behavior
Pylint says:
```
Undefined variable: '__builtins__'Python(undefined-variable)
Undefined variable: '__loader__'Python(undefined-variable)
```
### Expected behavior
No message at all.
### pylint --version output
pylint 2.4.4
astroid 2.3.3
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)]

**Comment by PCManticore** [comment]
I can't quite reproduce this. One way to test this locally is with:
```
from pylint.checkers.utils import is_builtin
print(is_builtin('__builtins__'))
print(is_builtin('__loader__'))
```

Both statements should return `True` . In case they don't, you can check if you have `__loader__` in your `builtins` module, with:
```
import builtins
builtins.__loader__
```

I am testing with the following versions which are installed from the master branch of both `pylint` and `astroid`:
```
pylint 2.5.0-dev1
astroid 2.4.0
Python 3.7.4 (default, Oct 18 2019, 14:15:28)
[Clang 8.1.0 (clang-802.0.42)]
```
